Sources/Host/POSIX/File.cpp

```diff
@@ -23,5 +24,31 @@ File::File(std::string const &path, uint32_t flags, uint32_t mode) {
 }
 
 File::~File() { ::close(_fd); }
+
+ErrorCode File::pread(std::string &buf, uint64_t &count, uint64_t offset) {
+  if (_fd < 0) {
```

Comment:
You should use your own `valid()` method, and just return `kErrorInvalidHandle` if it's `false`.